### PR TITLE
Manual Activity Entry

### LINKED ISSUE #49: Manual Activity Entry
**Is your feature request related to a problem? Please describe.**
I want to be able to manually enter meditations if I meditate with say YouTube that day

**Describe the solution you'd like**
On the stats screen, clicking a date on the calendar should bring up an Alert input that allows you to enter the amount of time you meditated in minutes. It should have 2 buttons, cancel and submit. Once submitted the activity should appear in the calendar.

**Additional context**
- TBD, the stats screen, screenshot
- One can use the Alert prompt to get the value https://reactnative.dev/docs/alert#prompt-ios

### PR DESCRIPTION
**Pull request by alexandrvicente**
## Description

Adds manual input to the stats calendar. I used react-native-paper's Dialog instead of Alert.prompt due to the lack of Android support.

## Ticket Link

Closes #49.

## How has this been tested?

Tested using these devices:

iPhone X 14.7.1
Pixel 2 API 30 - Simulator

## Screenshots

iOS Dark mode

Android Light mode

## Checklist

- [x] Added a "Closes [issue number]" in the ticket link section
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens

### CODE REVIEW

**Comment by watadarkstar** [comment]
Nice work! I will review the code tomorrow 🙂

File: redux/meditationSlice.ts [hunk 1]
```diff
@@ -27,6 +27,24 @@ const meditationSlice = createSlice({
...
+        timestamp: number
+        duration: number
+      }>
+    ) {
+      const { duration, timestamp } = action.payload
+
+      if (duration === 0) {
+        delete state.activity[timestamp]
+        return
+      }
```
Comment (watadarkstar) [inline redux/meditationSlice.ts:42]: If the duration is set to 0 it deletes the activity?

File: screens/Stats/Calendar/index.tsx [hunk 2]
```diff
@@ -20,10 +33,19 @@ export default function Calendar() {
     ...calendar,
   }
 
+  const onManualInput = ({ day, month, year }: DateObject) => {
+    const newTimestamp = new Date(year, month - 1, day).getTime()
```
Comment (watadarkstar) [inline screens/Stats/Calendar/index.tsx:37]: Why the `month - 1` here?

Comment (watadarkstar) [reply screens/Stats/Calendar/index.tsx:37]: Ah I see why it starts at zero

File: redux/meditationSlice.ts (hunk 1 above)
Comment (alexandrvicente) [reply redux/meditationSlice.ts:42]: Yeah, since there's no list to edit or delete afterwards I made the manual entry a single editable activity

Comment (watadarkstar) [reply redux/meditationSlice.ts:42]: We might want to make that clear in the UI but all good :)

**Review by watadarkstar — APPROVED** [review]
LGTM - some code comments could be useful to clear things up but all good :)

**Pull request merged**